src/components/dropdown/dropdown.ts

```diff
@@ -178,115 +183,94 @@ export default class IgcDropdownComponent
     return result;
   }
 
-  private handleKeyDown = (event: KeyboardEvent) => {
-    if (
-      this.open &&
-      event &&
-      (event.composedPath().includes(this.target) ||
-        event.composedPath().includes(this.content))
-    ) {
-      const key = event.key.toLowerCase();
-      const navKeys = [
-        'esc',
-        'escape',
-        'enter',
-        'arrowup',
-        'up',
-        'arrowdown',
-        'down',
-      ];
-      if (navKeys.indexOf(key) === -1) {
-        return;
-      }
+  protected setInitialSelection() {
+    const item = this.allItems.filter((item) => item.selected).at(-1);
+    this.allItems.forEach((item) => (item.selected = false));
+    if (item) {
+      this.selectItem(item, false);
+    }
+  }
+
+  protected handleKeyDown = (event: KeyboardEvent) => {
+    const path = event.composedPath();
+    if (!(path.includes(this.target) || path.includes(this.content))) return;
 
+    if (this.keyDownHandlers.has(event.key)) {
       event.preventDefault();
       event.stopPropagation();
-
-      switch (key) {
-        case 'esc':
-        case 'escape':
-          this.handleItemActionKey(DropdownActionKey.ESCAPE);
-          break;
-        case 'enter':
-          this.handleItemActionKey(DropdownActionKey.ENTER);
-          break;
-        case 'arrowup':
-        case 'up':
-          this.onArrowUpKeyDown();
-          break;
-        case 'arrowdown':
-        case 'down':
-          this.onArrowDownKeyDown();
-          break;
-        default:
-          return;
-      }
+      this.keyDownHandlers.get(event.key)?.call(this);
```

Comment:
Neat!
Assuming the alternative names were for old browsers, not sure why we even had them in for WC stuff to begin with.

Reply:
I believe the alternate names are actually down to OS architecture rather than browsers.

Reply:
The virtual keycodes are down to OS architecture. The `KeyboardEvent.key` should be unified according to [this](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#navigation_keys)

Reply:
Also
![image](https://user-images.githubusercontent.com/20106115/179167464-89b2976f-876f-4305-afd6-6f3ea0ba7dde.png)